context/config_type.go

```diff
@@ -0,0 +1,203 @@
+package context
+
+import (
+	"errors"
+	"fmt"
+
+	"gopkg.in/yaml.v3"
+)
+
+const defaultHostname = "github.com"
+const defaultGitProtocol = "https"
+
+// This interface describes interacting with some persistent configuration for gh.
+type Config interface {
```

Comment:
Do you think it would be worth extracting, at the end, any of the config stuff from `context` to a new package under `internal`?

Reply:
I do think that would be worthwhile. Having it `context` felt wrong and I wanted a `config` package; was just getting worried about how long this was taking. I'll see if I can do it quickly today.